http-ballerina/http_client_endpoint.bal

```diff
@@ -687,7 +687,13 @@ isolated function performDataBinding(Response response, TargetType targetType) r
         json payload = check response.getJsonPayload();
         return <map<json>> payload;
     } else if (targetType is typedesc<json>) {
-        return response.getJsonPayload();
+        json|ClientError payload = response.getJsonPayload();
+        if payload is json {
+            var result = payload.cloneWithType(targetType);
```

Comment:
When the `targetType` is json, here we do double clone. Any possibility to avoid it?